src/ip_data.py

```diff
@@ -155,25 +230,17 @@ def write_to_file(
             for record in interleave([targets, normal]):
                 writer.writerow(record)
 
+    def with_writer(self, writer_class: type[FileWriter]) -> Self:
+        self.writer_class = writer_class
+
+        return self
+
     def write_tables(self) -> Self:
-        self.write_to_db(
-            ACCESS_SCHEMA,
-            "access",
-            self.access_target(),
-            repeat(self.access_writer, NUM_ROWS),
-        )
-        self.write_to_db(
-            PEOPLE_SCHEMA,
-            "people",
-            self.people_target(),
-            repeat(self.people_writer, NUM_ROWS),
-        )
-        self.write_to_db(
-            SUBNET_SCHEMA,
-            "subnet",
-            self.subnet_target(),
-            repeat(self.subnet_writer, 0),
-        )
+        with self.writer_class(f"challenge_{self.dbid}") as writer:
+            writer.init_tables()
```

Comment:
why you you initing tables again when writing tables?

Reply:
intitialized when class is.